core/commands/swarm.go

```diff
@@ -61,6 +62,140 @@ const (
 	swarmDirectionOptionName = "direction"
 )
 
+var swarmPeeringCmd = &cmds.Command{
+	Helptext: cmds.HelpText{
+		Tagline: "modify the peering service.",
+		ShortDescription: `
+'ipfs swarm peering' is a tool to manupulate the peering service. 
+Peers in the peering service is maintained to be connected, reconnected 
+on disconnect with a back-off.
+`,
+	},
+	Subcommands: map[string]*cmds.Command{
+		"add": swarmPeeringAddCmd,
+		"ls":  swarmPeeringLsCmd,
+		"rm":  swarmPeeringRmCmd,
+	},
+}
+
+var swarmPeeringAddCmd = &cmds.Command{
+	Helptext: cmds.HelpText{
+		Tagline: "add peers into the peering service.",
+		ShortDescription: `
+'ipfs swarm peering add' adds peers into the peering service.
+`,
+	},
+	Arguments: []cmds.Argument{
+		cmds.StringArg("address", true, true, "address of peer to add into the PeeringService"),
+	},
+	Run: func(req *cmds.Request, res cmds.ResponseEmitter, env cmds.Environment) error {
+		addrs := make([]ma.Multiaddr, len(req.Arguments))
+
+		for i, arg := range req.Arguments {
+			addr, err := ma.NewMultiaddr(arg)
+			if err != nil {
+				return err
+			}
+
+			addrs[i] = addr
+		}
+
+		addInfos, err := peer.AddrInfosFromP2pAddrs(addrs...)
+		if err != nil {
+			return err
+		}
+
+		node, err := cmdenv.GetNode(env)
+		if err != nil {
+			return err
+		}
+
+		output := make([]string, len(addInfos))
+		for i, id := range addInfos {
+			output[i] = "add " + id.ID.Pretty()
+			node.Peering.AddPeer(id)
+
+			output[i] += " success"
+		}
+
+		return cmds.EmitOnce(res, &stringList{output})
```

Comment:
Our CLI is also our API, so we should strive to make this usable as an API. I'd

1. Introduce a `type peeringResult struct {ID peer.ID, Status string}`.
2. Emit each result separately (`res.Emit(peeringResult{id, "success"}`)
3. Format each result in a text encoder. This will be done _client-side_, allowing other users to receive nice json status results.